tutormfe/plugin.py

```diff
@@ -21,18 +21,29 @@
                 "production": {
                     "COACHING_ENABLED": "",
                     "ENABLE_DEMOGRAPHICS_COLLECTION": "",
+                    "APP_ID":"account",
                 },
             },
         },
         "GRADEBOOK_MFE_APP": {
             "name": "gradebook",
             "repository": "https://github.com/edx/frontend-app-gradebook",
             "port": 1994,
+            "env":{
+              "production":{
+                "APP_ID": "gradebook",
+                },
+            },
         },
         "LEARNING_MFE_APP": {
             "name": "learning",
             "repository": "https://github.com/edx/frontend-app-learning",
             "port": 2000,
+            "env":{
+              "production":{
+                "APP_ID": "learning",
```

Comment:
There is duplication here between `["name"]` and `["env"]["production"]["APP_ID"]`. They must always exactly equal one another or we will get bugs.

My PR [injects the app's `["name"]` directly into the prod overrides file](https://github.com/kdmccormick/tutor-mfe/pull/1/files#diff-f0ca4fd9435643b76a98d6d56134f4f2c3370a43633ea43dbf24080b6a69ee42R18) so that we do not need to duplicate the APP_ID here.

Reply:
I was thinking may be we only specify name as its, and then in Dockerfile we inject `APP_ID` at build time which will just replicate `name`.

Reply:
i.e 
```
######## {{ app["name"] }} (production)
FROM {{ app["name"] }}-dev AS {{ app["name"] }}
COPY ./env/production /openedx/env/production
RUN touch /openedx/env/production.override \
  {%- for key, value in app.get("env", {}).get("production", {}).items() %}
  && echo "{{ key }}='{{ value }}'" >> /openedx/env/production.override \
  {%- endfor %}
  && echo "done setting production overrides"
RUN echo "APP_ID={{ app["name"] }}"
RUN bash -c "set -a && source /openedx/env/production && source /openedx/env/production.override && npm run build"

```
The line before the last is the new added line

Reply:
I like that @ghassanmas , although I think you would need to set APP_ID in the same process as npm build, like:
```Dockerfile
RUN bash -c "set -a && source /openedx/env/production && source /openedx/env/production.override && APP_ID={{ app["name"] }} npm run build"
```

Reply:
+1 for a build-time override, and indeed you'd have to set it as a process ENV variable during the build.

Reply:
This should be resolved now if I understood both you correctly.

Reply:
Looks right to me 👍🏻